st2common/st2common/util/action_db.py

```diff
@@ -192,6 +192,12 @@ def update_liveaction_status(status=None, result=None, context=None, end_timesta
     LOG.debug('Updating ActionExection: "%s" with status="%s"', liveaction_db.id, status,
               extra=extra)
 
+    # If liveaction is already canceled, then do not allow status to be updated.
+    if liveaction_db.status == LIVEACTION_STATUS_CANCELED and status != LIVEACTION_STATUS_CANCELED:
```

Comment:
I guess this would probably indicate a programmer / user error or similar so throwing might be more correct (and propagate this error to the API / user if user tried to cancel it via API), but then again I could see the way everything it's currently implemented just making it a no-op if execution is already canceled is simpler and better.